src/foundry/foundry.js/pixi/containers/placeableObjects/token.d.ts

```diff
@@ -455,17 +384,21 @@ declare global {
      * @param effect  - The texture file-path of the effect icon to toggle on the Token.
      * @param active  - Force a certain active state for the effect
      * @param overlay - Whether to set the effect as the overlay effect?
+     *                  (default: `false`)
      * @returns Was the texture applied (true) or removed (false)
      */
     toggleEffect(
-      effect: string | ActiveEffect.Data,
+      effect: string | ConstructorParameters<typeof ActiveEffect>[0],
```

Comment:
I don't think the constructor parameter is the right thing here. I think you probably want the source data of an active effect. So something like `foundry.data.ActiveEffectData['_source']` or `foundry.documents.BaseActiveEffect['data']['_source']`.

I suppose what you are doing actually works because in the end, this will create a new `ActiveEffect`. But I think it's supposed to come from an existing effect (hence all the duplication with `deepClone` in `_toggleActiveEffect`

Reply:
It must not come from an `ActiveEffect`.
`token.toggleEffect(CONFIG.statusEffects[0])` is a valid call and will fail when using `foundry.data.ActiveEffectData['_source']`. But I exchanged `ActiveEffect` with `ConfiguredDocumentClassForName<'ActiveEffect'>` (which is, what foundry does).